libs/ml/include/ml/ops/divide.hpp

```diff
@@ -44,39 +45,81 @@ class Divide : public fetch::ml::Ops<T>
   virtual void Forward(VecTensorType const &inputs, ArrayType &output)
   {
     assert(inputs.size() == 2);
-    assert(inputs.at(0).get().shape() == inputs.at(1).get().shape());
     assert(inputs.at(0).get().shape() == output.shape());
 
-    fetch::math::Divide(inputs.at(0).get(), inputs.at(1).get(), output);
+    if ((inputs.at(0).get().shape() == inputs.at(1).get().shape()) ||
+        (inputs.at(1).get().size() > 1))
+    {  // array / array
+      fetch::math::Divide(inputs.at(0).get(), inputs.at(1).get(), output);
+    }
+    else
+    {  // array / scalar
```

Comment:
as it stands if you call this Forward with two arrays with different shapes (both with sizes > 1) it will trigger this else statement, and will divide the input(0) by the first element of input(1). That's a silent error so it's pretty dangerous.
Changing the if condition to simply (inputs(1).size() > 1) should fix it though